skbio/util/_decorator.py

```diff
@@ -82,3 +85,207 @@ def __get__(self, cls, owner):
 
     def __set__(self, obj, value):
         raise AttributeError("can't set attribute")
+
+
+class _state_decorator(object):
+    """ Base class for decorators of all public functionality.
+    """
+
+    _line_prefix = '\n        '
+    _required_kwargs = []
+
+    def _update_doc_string(self, func, state_desc,
+                           state_desc_prefix='State: '):
+        doc_lines = func.__doc__.split('\n')
+        # wrap lines at 79 characters, accounting for the length of
+        # self._line_prefix and start_desc_prefix
+        len_state_desc_prefix = len(state_desc_prefix)
+        wrap_at = 79 - (len(self._line_prefix) + len_state_desc_prefix)
+        state_desc_lines = wrap(state_desc, wrap_at)
+        # The first line of the state description should start with
+        # state_desc_prefix, while the others should start with spaces to align
+        # the text in this section. This is for consistency with numpydoc
+        # formatting of deprecation notices, which are done using the note
+        # Sphinx directive.
+        state_desc_lines[0] = '%s%s' % (state_desc_prefix, state_desc_lines[0])
+        header_spaces = ' ' * len_state_desc_prefix
+        for i, line in enumerate(state_desc_lines[1:], 1):
+            state_desc_lines[i] = '%s%s' % (header_spaces, line)
+
+        doc_lines.insert(
```

Comment:
This `insert` could be expensive for large docstrings O(n). Since this will be applied to every docstring in scikit-bio it's probably worth optimizing to avoid inserts at the beginning of the list.